File: MDR/src/mappers/UserMap.ts
import { Mapper } from "../core/infra/Mapper";

import { IUserDTO } from "../dto/IUserDTO";

import { User } from "../domain/User/User";

import { Model, Document } from "mongoose";
import { IUserPersistence } from "../dataschema/IUserPersistence";

// import RoleRepo from "../repos/roleRepo";

export class UserMap extends Mapper<User> {
  public static toDTO(user: User): IUserDTO {
    return {
      name: user.name.name,
      email: user.email.mail,
      pass: user.pass.value,
      sex: user.sex.sex,
      birth: user.birth.birth,
      address: user.address.address,
      phoneNumber: user.phoneNumber.pNumber,
      role: user.role.role,
    } as IUserDTO;
  }

  public static async toDomain(
    user: any | Model<IUserPersistence & Document>
  ): Promise<User> {
    const userOrError = User.create(user);

    userOrError.isFailure ? console.log(userOrError.error) : "Error!";

    return userOrError.isSuccess ? userOrError.getValue() : null;
  }

  public static toPersistence(user: User): any {
    return {
      name: user.name.name,
      email: user.email.mail,
      pass: user.pass.value,
      sex: user.sex.sex,
      birth: user.birth.birth,
      address: user.address.address,
      phoneNumber: user.phoneNumber.pNumber,
      role: user.role.role,
    };
  }
}

File: MDV/Mappers/UserMap.cs
using MDV.DTO.User;
using MDV.Domain.User;

namespace MDV.Map
{
    public class UserMap
    {
        public UserDTO DomainToDTO(User user){
            return new UserDTO
            {
                Id = user.Id.AsGuid(),
                Mail = user.Mail.mail,
                Pass = user.Pass.pass,
                Name = user.Name.name,
                Address = user.Address.address,
                PNumber = user.PNumber.number,
                Role = user.Role.role,
                Age = user.Age.date
            };
        }

        public User CreatingDTOToDomain(CreatingUserDTO userDTO){
                return new User(
                userDTO.Mail,
                userDTO.Pass,
                userDTO.Name,
                userDTO.Address,
                userDTO.PNumber,
                userDTO.Role,
                userDTO.Age);
        }
    }
}